plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/application/importoperation/GeneralProjectImportOperation.java

```diff
@@ -42,8 +41,9 @@ public GeneralProjectImportOperation(File projectDirectory) {
 
 		IWorkspace workspace = ResourcesPlugin.getWorkspace();
 		IProject project = workspace.getRoot().getProject(getProjectDirectory().getName());
-		overwriteExistingProject(project, workspace, monitor);
-		importToWorkspace(getProjectDirectory(), workspace, monitor);
+		if (overwriteExistingProject(project, workspace, monitor)) {
```

Comment:
This is a 2 step process

- first files are downloaded and user has control through the reuse repo flag
- then project description may be updated and this is controlled through this dialog